rllib/examples/documentation/checkpoints_and_exports.py

```diff
@@ -0,0 +1,261 @@
+    # and "agent2") map to which policies.
+    policy_mapping_fn=(
+        lambda agent_id, worker, episode, **kw: (
+            "pol1" if agent_id == "agent1" else "pol2"
+        )
+    ),
+    # Setting these is not necessary. All policies will be trained anyways by default.
+    policies_to_train=["pol1", "pol2"],
+)
+
+# Add the MultiAgentCartPole env to our config and build our Algorithm.
+my_ma_config.environment(
+    MultiAgentCartPole,
+    env_config={
+        "num_agents": 2,
+    },
+)
+
+my_ma_algo = my_ma_config.build()
+
+ma_checkpoint_dir = my_ma_algo.save()
+
+print(
+    "An Algorithm checkpoint has been created inside directory: "
+    f"'{ma_checkpoint_dir}'.\n"
+    "Individual Policy checkpoints can be found in "
+    f"'{os.path.join(ma_checkpoint_dir, 'policies')}'."
+)
+
+# __multi-agent-checkpoints-end__
+
+# __create-policy-checkpoint-begin__
+
+# Retrieve the Policy object from an Algorithm.
+# Note that for normal, single-agent Algorithms, the Policy ID is "default_policy".
+policy1 = my_ma_algo.get_policy(policy_id="pol1")
+
+# Tell RLlib to store an individual policy checkpoint (only for "pol1") inside
+# /tmp/my_policy_checkpoint
+policy1.export_checkpoint("/tmp/my_policy_checkpoint")
+
+# __create-policy-checkpoint-end__
+
+# __restore-policy-begin__
+
+import numpy as np  # noqa
+
+from ray.rllib.policy.policy import Policy  # noqa
+
+# Use the `from_checkpoint` utility of the Policy class:
+my_restored_policy = Policy.from_checkpoint("/tmp/my_policy_checkpoint")
+
+# Use the restored policy for serving actions.
+obs = np.array([0.0, 0.1, 0.2, 0.3])  # individual CartPole observation
+action = my_restored_policy.compute_single_action(obs)
+
+print(f"Computed action {action} from given CartPole observation.")
+
+# __restore-policy-end__
+
+
+# __restore-algorithm-from-checkpoint-with-fewer-policies-begin__
+
+from ray.rllib.algorithms.ppo import PPOConfig  # noqa
+from ray.rllib.examples.env.multi_agent import MultiAgentCartPole  # noqa
+
+# Set up an Algorithm with 5 Policies.
+algo_w_5_policies = (
+    PPOConfig()
+    .environment(
+        env=MultiAgentCartPole,
+        env_config={
+            "num_agents": 5,
+        },
+    )
+    .multi_agent(
+        policies={"pol0", "pol1", "pol2", "pol3", "pol4"},
+        # Map "agent0" -> "pol0", etc...
+        policy_mapping_fn=(
+            lambda agent_id, episode, worker, **kwargs: f"pol{agent_id}"
+        ),
+    )
+    .build()
+)
+
+# .. train one iteration ..
+algo_w_5_policies.train()
+# .. and call `save()` to create a checkpoint.
+path_to_checkpoint = algo_w_5_policies.save()
+print(
+    "An Algorithm checkpoint has been created inside directory: "
+    f"'{path_to_checkpoint}'. It should contain 5 policies in the 'policies/' sub dir."
+)
+# Let's terminate the algo for demonstration purposes.
+algo_w_5_policies.stop()
+
+# We will now recreate a new algo from this checkpoint, but only with 2 of the
```

Comment:
I think stylistically, whenever you have whole paragraphs embedded in code, it's usually better to have multiple `literalinclude` statements instead (just for the code) and add these paragraphs to your `rst` file instead. The good thing is that I don't think this is really any more work at all, just takes getting used to.

Reply:
I split this up a little.